src/vs/platform/lifecycle/common/lifecycle.ts

```diff
@@ -20,7 +20,22 @@ export const ILifecycleService = createDecorator<ILifecycleService>('lifecycleSe
  */
 export interface ShutdownEvent {
 	veto(value: boolean | TPromise<boolean>): void;
-	quitRequested: boolean;
+	reason: ShutdownReason;
+}
+
+export enum ShutdownReason {
+
+	/** Window is closed */
+	CLOSE,
+
+	/** Application is quit */
+	QUIT,
+
+	/** Window is reloaded */
+	RELOAD,
+
+	/** Other configuration loaded into window */
```

Comment:
@Tyriar basically opening another folder into an existing window or closing a folder. anything that changes the URL that is associated with a window to something else is a `LOAD` (whereas `RELOAD` loads the same URL).